packages/miniflare/src/plugins/pipelines/index.ts

```diff
@@ -0,0 +1,104 @@
+import { SharedBindings } from "miniflare:shared";
+import SCRIPT_PIPELINE_OBJECT from "worker:pipelines/pipeline";
+import { z } from "zod";
+import {
+	kVoid,
+	Service,
+	Worker_Binding_DurableObjectNamespaceDesignator,
+} from "../../runtime";
+import {
+	getMiniflareObjectBindings,
+	namespaceKeys,
+	objectEntryWorker,
+	Plugin,
+	ProxyNodeBinding,
+	SERVICE_LOOPBACK,
+} from "../shared";
+
+export const PipelineOptionsSchema = z.object({
+	pipelines: z.union([z.record(z.string()), z.string().array()]).optional(),
+});
+
+export const PIPELINES_PLUGIN_NAME = "pipelines";
+const SERVICE_PIPELINE_PREFIX = `${PIPELINES_PLUGIN_NAME}:pipeline`;
+const PIPELINE_OBJECT_CLASS_NAME = "Pipeline";
+const PIPELINE_OBJECT: Worker_Binding_DurableObjectNamespaceDesignator = {
+	serviceName: SERVICE_PIPELINE_PREFIX,
+	className: PIPELINE_OBJECT_CLASS_NAME,
+};
+
+export const PIPELINE_PLUGIN: Plugin<typeof PipelineOptionsSchema> = {
+	options: PipelineOptionsSchema,
+	getBindings(options) {
+		const pipelines = bindingEntries(options.pipelines);
+		return pipelines.map<Service>(([name]) => ({
+			name,
+			service: { name: SERVICE_PIPELINE_PREFIX },
```

Comment:
think this needs the id?
```suggestion
			service: { name: `SERVICE_PIPELINE_PREFIX:${id}` },
```

Reply:
I don't think that is needed here since these are not backed by DOs, but JSPRC bindings. @penalosa do you recall if that is correct?

https://github.com/cloudflare/workers-sdk/blob/9560ae5fed94d1858e380804c903c5020318501d/packages/miniflare/src/workers/pipelines/pipeline.worker.ts#L3

Reply:
so it looks like your set up is 
user worker ---binding--> pipeline entry worker --> pipeline service worker

where the pipeline entryworker is the `services` declared in line 49 with `objectEntryWorker()`
so those are per pipeline binding and have an id in their name, so when you use `getBinding` miniflare is hooking up user worker -> pipeline entry worker

along the lines of d1 and kv in this diagram:

![image](https://github.com/user-attachments/assets/13fcb6bb-bdc2-4328-b348-d519f37a3c05)

Reply:
I am seeing this error in the tests now (was passing before without ID param). Did I miss a part in [5110fca](https://github.com/cloudflare/workers-sdk/pull/7950/commits/5110fcaa4c3752d8267745dcc27a80d15e3f0623)?

```
workerd/server/server.c++:2009: warning: A DurableObjectNamespace in the config referenced the class "Pipeline", but no such Durable Object class is exported from the worker. Please make sure the class name matches, it is exported, and the class extends 'DurableObject'. Attempts to call to this Durable Object class will fail at runtime, but historically this was not a startup-time error. Future versions of workerd may make this a startup-time error.
Worker "core:user:"'s binding "PIPELINE" refers to a service "SERVICE_PIPELINE_PREFIX:PIPELINE", but no such service is defined.
```